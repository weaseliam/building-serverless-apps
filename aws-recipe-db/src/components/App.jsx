import React from 'react';
import { connect } from 'react-redux';
import { Route, Switch } from 'react-router-dom';

import HeaderView from './HeaderView';
import LeftView from './LeftView';
import RecipeDetails from './RecipeDetails';
import RecipeEdit from './RecipeEdit';
import PageNotFound from './PageNotFound';
import RecipeHome from './RecipeHome';
import Login from './Login';
import Signup from './Signup';

const App = (props) => {
  return (
    <div className="container">
      <div className="row">
        <HeaderView title="Recipe Database" />
      </div>
      <div className="row">
        <div className="col-xs-4">
          <LeftView />
        </div>
        <div className="col-xs-8">
          <Switch>
            <Route exact path="/" component={RecipeHome} />
            <Route exact path="/login" component={Login} />
            <Route exact path="/signup" component={Signup} />
            <Route exact path="/edit" component={RecipeEdit} />
            <Route exact path="/recipes/:id" component={RecipeDetails} />
            <Route component={PageNotFound} />
          </Switch>
        </div>
      </div>
    </div>
  );
};

export default connect(
  state => state.routing
)(App);
