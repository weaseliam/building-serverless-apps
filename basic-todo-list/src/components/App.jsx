import React, { Component } from 'react';

import Header from './Header';
import TodoList from './TodoList';
import TodoInput from './TodoInput';

class App extends Component {
  constructor() {
    super();

    this.state = {
      todos: [{
        name: "Hello World !",
        done: false
      }]
    };
  }

  handleAddTodo = (name) => {
    let todo = {
      name,
      done: false
    };
    let todos = Object.assign(this.state.todos);
    todos.push(todo);

    this.setState({
      todos
    });
  }

  handleToggleTodo = (todo1, index) => {
    console.log(`Toggle todo for index ${index}`);

    let todos = Object.assign(this.state.todos);
    let todo = todos[index];
    todo.done = !todo.done;

    this.setState({
      todos
    });
  }

  render() {
    const { todos } = this.state;

    return (
      <div>
        <Header />
        <TodoList todos={todos} onToggleTodo={this.handleToggleTodo} />
        <TodoInput onAddTodo={this.handleAddTodo} />
      </div>
    );
  }
}

export default App;
