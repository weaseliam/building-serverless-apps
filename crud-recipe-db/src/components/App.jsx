import React, { Component } from 'react';

import HeaderView from './HeaderView';
import LeftView from './LeftView';
import RightView from './RightView';

class App extends Component {
  constructor() {
    super();

    this.state = {
      recipes: [],
      showEdit: false,
      recipeToShow: null,
      query: null
    };
  }

  handleSubmitRecipe = (recipe) => {
    let recipes = Object.assign(this.state.recipes);
    recipes.push(recipe);

    this.setState({
      recipes,
      recipeToShow: recipe,
      showEdit: false
    });
  }

  handleCreateRecipe = () => {
    this.setState({
      showEdit: true
    });
  }

  handleSelectRecipe = (recipe) => {
    this.setState({
      showEdit: false,
      recipeToShow: recipe
    });
  }

  handleSearch = (value) => {
    this.setState({
      query: value
    });
  }

  render() {
    let { recipes, query } = this.state;
    let filteredRecipes = query ? 
      recipes.filter((recipe) => 
        query && recipe.name.toLowerCase().indexOf(query.toLowerCase()) > -1) : 
      recipes;

    return (
      <div className="container">
        <div className="row">
          <HeaderView title="Recipe Database" />
        </div>
        <div className="row">
          <div className="col-xs-4">
            <LeftView
              recipes={filteredRecipes}
              onCreateRecipe={this.handleCreateRecipe}
              onSelect={this.handleSelectRecipe}
              onSearch={this.handleSearch} />
          </div>
          <div className="col-xs-8">
            <RightView
              onSubmitRecipe={this.handleSubmitRecipe}
              showEdit={this.state.showEdit}
              recipe={this.state.recipeToShow} />
          </div>
        </div>
      </div>
    );
  }
}

export default App;
