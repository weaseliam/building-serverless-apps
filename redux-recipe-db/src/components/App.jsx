import React from 'react';
import { connect } from 'react-redux';
import { Route, Switch } from 'react-router-dom';

import HeaderView from './HeaderView';
import LeftView from './LeftView';
import RecipeDetails from './RecipeDetails';
import RecipeEdit from './RecipeEdit';
import PageNotFound from './PageNotFound';
import RecipeHome from './RecipeHome';

const App = (props) => {
  return (
    <div className="container">
      <div className="row">
        <HeaderView title="Recipe Database" />
      </div>
      <div className="row">
        <div className="col-xs-4">
          <LeftView />
        </div>
        <div className="col-xs-8">
          <Switch>
            <Route exact path="/" component={RecipeHome} />
            <Route exact path="/edit" component={RecipeEdit} />
            <Route exact path="/recipes/:id" component={RecipeDetails} />
            <Route component={PageNotFound} />
          </Switch>
        </div>
      </div>
    </div>
  );
};

export default connect(
  state => state.routing
)(App);
