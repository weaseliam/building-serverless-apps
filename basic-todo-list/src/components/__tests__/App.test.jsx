import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';

test('App renders as expected', () => {
  const component = renderer.create(<App />);

  expect(component.toJSON()).toMatchSnapshot();
});
