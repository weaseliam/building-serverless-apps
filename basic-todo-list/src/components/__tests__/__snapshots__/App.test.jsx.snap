// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders as expected 1`] = `
<div>
  <h1>
    Todo List
  </h1>
  <ul>
    <li
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      Hello World !
    </li>
  </ul>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add
    </button>
  </form>
</div>
`;
